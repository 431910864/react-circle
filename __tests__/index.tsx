import * as React from 'react';
import { shallow, render, mount } from 'enzyme';
import Circle, {CircleProps} from '../src';

describe('ReactCircle', () => {
  const setup = (props?: Partial<CircleProps>) => {
    const initialOptions = {
      progress: 25
    }
  
    const circle = shallow(
      <Circle {...initialOptions} {...props} />
    );
    
    return {
      circle
    };
  };

  it('Should render with 10px line', () => {
    const {circle} = setup({lineWidth: '10'});
    expect(circle.find('circle').first().prop('strokeWidth')).toBe('10')
  })

  it('Should render inside circle of same width as outside one', () => {
    const {circle} = setup({lineWidth: '10'});
    const innerWidth = circle.find('circle').first().prop('strokeWidth');
    const outerWidth = circle.find('circle').last().prop('strokeWidth');
    expect(innerWidth).toBe(outerWidth)
  })

  it('Should render a with no text', () => {
    const {circle} = setup({showPercentage: false});
    expect(circle.find('text').exists()).toBeFalsy();
  })

  it('Should render a text with 25%', () => {
    const {circle} = setup();
    expect(circle.find('text').text()).toEqual('25%');
  });

  it('Should render in default with a square stroke', () => {
    const { circle } = setup();
    expect(circle.find('circle').last().prop('strokeLinecap')).toBe('butt');
});

  it('Should render with a rounded stroke', () => {
    const {circle} = setup({roundedStroke: true});
   expect(circle.find('circle').last().prop('strokeLinecap')).toBe('round');
  });


  it('Should default render with animation', () => {
    const { circle } = setup({ animate: false });
    const innerCircle = circle.find('circle').last()
    expect(innerCircle.prop('style')).toMatchSnapshot();
  })

  it('Should render without animation', () => {
    const { circle } = setup({animate:false});
    const innerCircle = circle.find('circle').last()
    expect(innerCircle.prop('style')).toMatchSnapshot();
  })
});
