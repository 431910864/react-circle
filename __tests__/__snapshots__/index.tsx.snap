// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactCircle Should default render with animation 1`] = `
Object {
  "strokeDashoffset": 825,
  "transition": null,
}
`;

exports[`ReactCircle Should render without animation 1`] = `
Object {
  "strokeDashoffset": 825,
  "transition": null,
}
`;
